programs/lz4cli.c

```diff
@@ -693,7 +692,19 @@ int main(int argc, const char** argv)
         break;
     }
 
-    if (multiple_inputs==0 && mode != om_list) assert(output_filename);
+    if (mode == om_list){
+        /* Exit if trying to read from stdin as this isn't supported in this mode */
+        if(!strcmp(input_filename, stdinmark)){
```

Comment:
May be better to refuse read from non-seekable file, instead of just testing for stdin; this will stop things like
```sh
lz4 --list < a-regular-file.lz4
```
from working.

Reply:
Isn't this already the case ?

```
./lz4 --list < a-regular-file.lz4
refusing to read from standard input in --list mode
```

Reply:
In this example, the shell redirected a regular file as stdin (fd 0) for the new process; this fd can be seeked because it was a regular file.